transcom/mymove
# MAIN B-20603 backend boat office api

## [B-20603](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20603)

INT PR: https://github.com/transcom/mymove/pull/13521

## Summary

This is to add/modify services and ghc api endpoints related to Boat Shipment for office users.

- Added validation - `isRoadworthy` should be null if `hasTrailer` is false.
- Added boat shipment to MTOShipment ghc API(GET, POST, PATCH).

### How to test
1. Create a move with any shipment.
2. As an office user ->  Go to the move details page for the move you have created.
3. Use the browser dev tools get the info you need for the Swagger or Postman.
4. Use the POST to create a new move and refresh the move details page, and you should see a new boat shipment created.

Use Swagger or Postman to check ghc APIs:

Import `ghc.yaml` file from swagger folder if you have not already.

POST - officelocal:3000{{baseUrl}}/mto-shipments
http://officelocal:3000/swagger-ui/ghc.html#/mtoShipment/createMTOShipment
Example body to use:
```
{
  "moveTaskOrderID": "a615abd5-c64a-4c0d-b438-2b5e66790130",
  "shipmentType": "BOAT_TOW_AWAY",
  "boatShipment": {
    "type": "TOW_AWAY",
    "year": 1111,
    "model": "TOW_AWAY",
    "make": "TOW_AWAY",
    "lengthInInches": 169,
    "widthInInches": 83,
    "heightInInches": 78,
    "hasTrailer": false
  }
}
```

PATCH - officelocal:3000{{baseUrl}}/move_task_orders/:moveTaskOrderID/mto_shipments/:shipmentID
http://officelocal:3000/swagger-ui/ghc.html#/mtoShipment/updateMTOShipment
Example body to use:
```
{
  "moveTaskOrderID": "2ceb3116-b24b-41df-a023-67ac09ec13de",
  "shipmentType": "BOAT_HAUL_AWAY",
  "boatShipment": {
    "type": "HAUL_AWAY"
  }
}
```

GET - http://officelocal:3000/swagger-ui/ghc.html#/mtoShipment/listMTOShipments
GET - http://officelocal:3000/swagger-ui/ghc.html#/mtoShipment/getShipment

## Screenshots
![image](https://github.com/user-attachments/assets/d34b48f1-a4b6-4539-b20a-f91d10f0e8f4)
![image](https://github.com/user-attachments/assets/423c2cc6-cc73-4114-9a18-1af1618aae23)
![image](https://github.com/user-attachments/assets/e5e220ff-4996-4455-8a3b-216edcb2a3f2)
![image](https://github.com/user-attachments/assets/cac24958-08c3-4c16-8b98-e0bc64470d9d)
